elfo-logger/src/filtering_layer.rs

```diff
@@ -76,6 +79,32 @@ impl<S: Subscriber> Layer<S> for FilteringLayer {
         // We don't need to recheck `.targets` here, because `.register_callsite()`
         // would already eliminate logs that would be filtered by it.
         let level = *meta.level();
+
+        // `tracing-log` doesn't call `.register_callsite()`, so we have to recheck
+        // `.targets` in this one case.
+        let name = meta.name();
+        if let Some(&log_name) = self.inner.log_metadata_name.get() {
+            // We've already got logs from `tracing-log`, just compare str
+            // pointers for performance.
+            if std::ptr::eq(log_name, name) {
+                let config = self.inner.config.load();
+                if !config.targets.would_enable(meta.target(), meta.level()) {
+                    return false;
+                }
+            }
+        } else if name == "log record" {
```

Comment:
If the `tracing-log` feature is disabled, we shouldn't check and even use `log_metadata_name` at all.